build.mk

```diff
@@ -0,0 +1,88 @@
+OS = $(shell uname | tr A-Z a-z)
+GOARCH = $(shell go env GOARCH)
+
+## The expected golang version; crashes if the local env is different
+GOLANG_VERSION ?= 1.18
+
+## Build variables
+BUILD_DIR ?= bin
+BINARY_NAME_client ?= p1
+BINARY_NAME_pocket ?= pocket
+POST_BUILD_TARGETS = rename-binaries
+VERSION ?= $(shell git describe --tags --exact-match 2>/dev/null || git symbolic-ref -q --short HEAD)
+COMMIT_HASH ?= $(shell git rev-parse --short HEAD 2>/dev/null)
+DATE_FMT = +%FT%T%z
+## Check if the distribution has configured a SOURCE_DATE_EPOCH env var in order to generate a reproducible output. (https://reproducible-builds.org/docs/source-date-epoch/)
+ifdef SOURCE_DATE_EPOCH
+    BUILD_DATE ?= $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u "$(DATE_FMT)")
+else
+    BUILD_DATE ?= $(shell date "$(DATE_FMT)")
+endif
+LDFLAGS += -X app.AppVersion=${VERSION} -X app.CommitHash=${COMMIT_HASH} -X app.BuildDate=${BUILD_DATE}
```

Comment:
1. Do we not need to add variables in the go source code corresponding to this?

2. Per our discussion in discord, can you add a comment why we're not using `debug.ReadBuildInfo()`. Doesn't have to be long, but future maintainers/readers will know we looked into it.

Reply:
Comment and variables added to the correct package :)